### PR TITLE
Fix taurus dep

### LINKED ISSUE #7: Remove LICENSE from recipe
The LICENSE is now part of the pypi package.
We should think about removing it from the recipe when releasing a new version.

Creating this issue to not forget.

### PR DESCRIPTION
**Pull request by beenje**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Some users reported issue when installing latest sardana (3.1.3) with conda as it pulled taurus 5. I updated the requirements to force taurus `<5`.

I took the opportunity to remove LICENSE from recipe (Fix #7)

### CODE REVIEW

**Comment by beenje** [comment]
@conda-forge-admin, please rerender

**Comment by cpascual** [comment]
> Some users reported issue when installing latest sardana (3.1.3) with conda as it pulled taurus 5.

Hi @beenje , do you have a link to those issues?
Without knowing them, I can imagine 2 alternative scenarios:

- a) taurus 5 is incompatible with sardana 3.1.3 for some reason 
- b) the user prefers to keep taurus at version 4 because some other dependency (not sardana) is not ready for taurus 5

In the "a)" case, I would agree on this MR, but only if it cannot be fixed with a hotfix to either Taurus or Sardana
In the b) case, I think it is responsibility of the user to force an explicit restriction for the desired taurus version when installing sardana (e.g. conda install sardana taurus<5)

In other words: IMHO it is better to allow sardana to depend on latest taurus, whenever possible.

**Review by cpascual — COMMENTED** [review]
> Some users reported issue when installing latest sardana (3.1.3) with conda as it pulled taurus 5.

Hi @beenje , do you have a link to those issues?
Without knowing them, I can imagine 2 alternative scenarios:

- a) taurus 5 is incompatible with sardana 3.1.3 for some reason 
- b) the user prefers to keep taurus at version 4 because some other dependency (not sardana) is not ready for taurus 5

In the "a)" case, I would agree on this MR, but only if it cannot be fixed with a hotfix to either Taurus or Sardana
In the b) case, I think it is the responsibility of the user to force an explicit restriction for the desired taurus version when installing sardana (e.g. `conda install sardana taurus<5`)

In other words: IMHO it is better to allow sardana to depend on latest taurus, whenever possible.

**Comment by reszelaz** [comment]
Hi @cpascual,
The issue is here: https://gitlab.com/sardana-org/sardana/-/issues/1719#note_770990612.
So, I think this is a). And following your suggestion we need to decide which: Taurus or Sardana requires a hotfix.

**Comment by beenje** [comment]
Indeed this is an incompatibility between sardana 3.1.3 and taurus 5.

My understanding is that `taurus.core.release.version` had been deprecated for some time.
So I'd vote to patch sardana.

I'd also suggest to use [importlib-metadata](https://importlib-metadata.readthedocs.io/) to get dependencies version.
That would avoid relying on packages internal implementation.

**Comment by cpascual** [comment]
Ideally I would prefer this fixed in taurus but, after looking at it, I do not see any way of fixing it without reversing the main improvement regarding imports in taurus 5.

Therefore, @reszelaz  and I agreed to hotfix this in sardana (probably using importlib-metadata, as suggested by @beenje )

**Comment by beenje** [comment]
In the meantime, should we merge this so that installing this version from conda won't pull Taurus 5?

**Comment by cpascual** [comment]
done
Thanks @beenje ! 

(We should remember to revert when the hotfix is ready)

**Comment by reszelaz** [comment]
> Therefore, @reszelaz and I agreed to hotfix this in sardana (probably using importlib-metadata, as suggested by @beenje )

Sardana 3.2 is very closed to be released. I think it does not make sense to release hotfix 3.1.4 cause the 3.2 already uses the compatible way to check the taurus version.
